packages/playwright-core/src/server/network.ts

```diff
@@ -265,9 +265,18 @@ export class Route extends SdkObject {
         isBase64 = false;
       }
     }
+    const headers = [...(overrides.headers || [])];
+    if (overrides.cors !== false) {
+      const corsHeader = headers.find(({ name, value }) => name === 'access-control-allow-origin');
+      // See https://github.com/microsoft/playwright/issues/12929
+      if (!corsHeader) {
+        const origin = this._request.headerValue('origin') || '*';
+        headers.push({ name: 'access-control-allow-origin', value: origin });
```

Comment:
MDN suggests to also send `Vary: Origin`: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#access-control-allow-origin.